components/kyma-environment-broker/internal/process/deprovisioning/event_hub_step.go

```diff
@@ -0,0 +1,145 @@
+package deprovisioning
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/hyperscaler"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/hyperscaler/azure"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/process"
+	processazure "github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/process/azure"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/storage"
+)
+
+type DeprovisionAzureEventHubStep struct {
+	OperationManager *process.DeprovisionOperationManager
+	operationStorage storage.Operations
+	instanceStorage  storage.Instances
+	processazure.EventHub
+}
+
+func NewDeprovisionAzureEventHubStep(os storage.Operations, is storage.Instances, hyperscalerProvider azure.HyperscalerProvider, accountProvider hyperscaler.AccountProvider, ctx context.Context) DeprovisionAzureEventHubStep {
+	return DeprovisionAzureEventHubStep{
+		OperationManager: process.NewDeprovisionOperationManager(os),
+		operationStorage: os,
+		instanceStorage:  is,
+		EventHub: processazure.EventHub{
+			HyperscalerProvider: hyperscalerProvider,
+			AccountProvider:     accountProvider,
+			Context:             ctx,
+		},
+	}
+}
+
+var _ Step = (*DeprovisionAzureEventHubStep)(nil)
+
+func (s DeprovisionAzureEventHubStep) Name() string {
+	return "Deprovision Azure Event Hubs"
+}
+
+func (s DeprovisionAzureEventHubStep) Run(operation internal.DeprovisioningOperation, log logrus.FieldLogger) (internal.DeprovisioningOperation, time.Duration, error) {
+	if operation.EventHub.Deleted {
+		log.Info("Event Hub is already deprovisioned")
+		return operation, 0, nil
+	}
+
+	hypType := hyperscaler.Azure
+
+	instance, err := getInstance(s.instanceStorage, operation, log)
+	switch err.(type) {
+	case nil:
+	case instanceNotFoundError:
+		operation.EventHub.Deleted = true
+		updatedOperation, err := s.operationStorage.UpdateDeprovisioningOperation(operation)
+		log.Errorf("while updating the database, error: %v", err)
+		return s.OperationManager.OperationSucceeded(*updatedOperation, "instance already deprovisioned")
+	default:
+		return operation, 1 * time.Second, nil
```

Comment:
Should this be `return operation, 1 * time.Second, err`?

Reply:
@nachtmaar What do you think?
What is the difference between returning an error and `nil` (as well as returning the duration)

Reply:
I think to repeat the operation, the `error` should be `nil`